### PR TITLE
Wagtail 6.1 Upgrade

### PR DESCRIPTION
**Pull request by nickmoreton**
[Support Ticket](https://torchbox.monday.com/boards/1318157635/pulses/1521862197)

### Description of Changes Made

When merged this will upgrade Wagtail to v6.1

# Wagtail 6.1 upgrade check list
Add a link to the commit or file for each consideration if you made any code changes.
## Code reviewing a consideration.
Use the tick box to indicate a consideration has been code reviewed as OK

### Changes affecting wagtail customizations
  - [x] [Changes to submissionslistview class](https://docs.wagtail.org/en/stable/releases/6.1.html##changes-to-submissionslistview-class)
  - [x] [Register_user_listing_buttons hook signature changed](https://docs.wagtail.org/en/stable/releases/6.1.html##register-user-listing-buttons-hook-signature-changed)
  - [x] [Password_required_template has changed to wagtail_password_required_template](https://docs.wagtail.org/en/stable/releases/6.1.html##password-required-template-has-changed-to-wagtail-password-required-template) [**Updated Here**](https://github.com/torchbox/torchbox.com/pull/262/commits/e3e291a3add2e1bd0ec5acd0f843804c54e4145b)
  - [x] [Document_password_required_template has changed to wagtaildocs_password_required_template](https://docs.wagtail.org/en/stable/releases/6.1.html##document-password-required-template-has-changed-to-wagtaildocs-password-required-template)
### Changes to undocumented internals
  - [x] [Deprecation of user_listing_buttons template tag](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-user-listing-buttons-template-tag)
  - [x] [Deprecation of wagtailusers_groups:users url pattern](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-wagtailusers-groups-users-url-pattern)
  - [x] [Deprecation of window.chooserurls within draftail choosers](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-window-chooserurls-within-draftail-choosers)
  - [x] [Deprecation of window.initblockwidget to initialize a streamfield block](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-window-initblockwidget-to-initialize-a-streamfield-block)
  - [x] [Removal of jquery from base client-side widget and boundwidget classes](https://docs.wagtail.org/en/stable/releases/6.1.html##removal-of-jquery-from-base-client-side-widget-and-boundwidget-classes)
  - [x] [Window.urlify deprecated](https://docs.wagtail.org/en/stable/releases/6.1.html##window-urlify-deprecated)
  - [x] [Window.xregexp polyfill removed](https://docs.wagtail.org/en/stable/releases/6.1.html##window-xregexp-polyfill-removed)
### Other changes:
- Added html5lib [here](https://github.com/torchbox/torchbox.com/pull/262/commits/0bd5d9cc7ced61ed771ae5ab72942b6f464ea6e9) as it's no longer a dependency of Wagtail core and is used here: https://github.com/torchbox/torchbox.com/blob/0bd5d9cc7ced61ed771ae5ab72942b6f464ea6e9/tbx/core/templatetags/reading_time_tags.py#L37
- Because Beautifulsoup4 is used in this site I have also included that dependency [here](https://github.com/torchbox/torchbox.com/pull/262/commits/0bd5d9cc7ced61ed771ae5ab72942b6f464ea6e9) as a safety measure

### Screenshots

<details>
  <summary>Expand to see more</summary>

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [ ] Updated field spec (https://docs.google.com/spreadsheets/d/1v1yqcSC54Vag4mxmTp8e6ZrnxITqzbjCA4A9XPQdXk8/edit?usp=sharing)
- [ ] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [x] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] HTML validation passes
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -41,14 +41,33 @@ As indicated [here](./custom-features/migration-friendly-streamfields.md), this
...
-### Code blocks using wagtail-markdown
+#### Code blocks using wagtail-markdown
 
 - Check that [code blocks](front-end/markdown-codehilite.md) still display as expected
 
+#### Wagtailmedia
+
+- Check if the [wagtailmedia] package has been updated and if possible use the official release.
+
+Currently using https://github.com/torchbox-forks/wagtailmedia.git @ 0.15.2+tbx
```
[the remark this answers is not in the document]
Comment (nickmoreton) [reply docs/upgrading.md:69]: Good spot. Those changes won't make it to our fork until they are merged and we can pull the changes in.

I think we can wait for a short while and once it's released we can use the correct version.

[the remark this answers is not in the document]
Comment (zerolab) [reply docs/upgrading.md:69]: > those changes won't make it to our fork until they are merged and we can pull the changes in.

when they get merged, there will be a new release, so there should not be a need for another fork.

the "when" is tomorrow

[the remark this answers is not in the document]
Comment (nickmoreton) [reply docs/upgrading.md]: Thanks @zerolab The wagtailmedia dependency is now updated to use the official release (0.15.2) https://github.com/torchbox/torchbox.com/pull/262/commits/54b4f6805f13a6c382b26c8cefd29094ffc747d2

**Review by alxbridge — APPROVED** [review]
@nickmoreton It looks like you've addressed all of Victor's feedback - thanks. I've just pushed one more dependency bump, for a security patch of `urllib3` that was released yesterday. With that, I'm happy to approve this PR.

**Pull request merged**